File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Levels/TimeLevelDto.cs
namespace SoccerPlayerApi.Dtos.Levels;

public class TimeLevelDto
{
    public int Id { get; set; }

    public string Label { get; set; }


    public int? AncestorId { get; set; }
    public TimeLevelDto? Ancestor { get; set; }
}

File: SoccerPlayerApi/clientapp/src/app/models/levels/TimeLevelDto.ts
export class TimeLevelDto {
    id: number;
    label: string;

    ancestorId?: number;
    ancestor?: TimeLevelDto;

    constructor(id: number, label: string, ancestorId?: number, ancestor?: TimeLevelDto) {
        this.id = id;
        this.label = label;
        this.ancestorId = ancestorId;
        this.ancestor = ancestor;
    }
}
